woocommerce/woocommerce-ios
# [Coupons in Orders]Provide a better handling for adding coupons to orders errors.

Closes: #10041


## Description

With this PR we provide a better handling for errors when adding coupons to orders. We do this through two tasks:

1. Disable the Add Coupon row in the Order Details screen if there are no products, as it will surely fail.
2. Show a proper message to the user and reset the Order coupon if adding the coupon fails because it cannot be added to the order.

### Known problem

Currently, if we don't handle the error when we sync an order, we show `Unable to load taxes for order`, which might not be the real problem behind the error. This however shouldn't affect us here as we handle the error.

## Testing instructions


1. Go to orders and tap to create a new one
2. Add a product
3. Try to add a coupon not applicable to the added product. It should fail with a specific error. The coupon should be removed from the order.
4. Add a coupon applicable to the product. It should sync succesfully.

## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/1864060/2c3b5a3d-9754-4487-b50f-0318199bfa2d

---
